python/dlr/dlr_model.py

```diff
@@ -436,12 +472,20 @@ def get_input(self, name, shape=None):
             raise ValueError('Since set_input() was never called with ' +
                              'input {}, we cannot infer its shape. '.format(name) +
                              'Shape parameter should be explicitly specified')
+        # default dtype and ctype is float32
+        data_dtype = np.float32
+        data_ctype = ctypes.c_float
+        if name not in self.weight_names:
+            index = self._get_input_index(name)
+            model_input_dtype = self.get_input_dtype(index)
+            if model_input_dtype in QUANT_TYPES:
+                data_dtype, data_ctype = QUANT_TYPES[model_input_dtype]
```

Comment:
It looks this chunk of code is repeated. We can create a helper function for it.